src/renderers/shared/fiber/ReactFiberUpdateQueue.js

```diff
@@ -31,14 +30,14 @@ type PartialState<State, Props> =
 // Callbacks are not validated until invocation
 type Callback = mixed;
 
-export type Update = {
+export type Update<State> = {
   expirationTime: ExpirationTime,
   partialState: PartialState<any, any>,
   callback: Callback | null,
   isReplace: boolean,
   isForced: boolean,
-  isTopLevelUnmount: boolean,
-  next: Update | null,
+  next: Update<State> | null,
+  nextCallback: Update<State> | null,
```

Comment:
If you made the callback list an array, we wouldn't need this field and we'd also save one field on the UpdateQueue.

We use a lot of linked lists elsewhere for legit reasons but we probably shouldn't have that leak to everything.